File: fixtures/extraction/r/test_roles/test_source.r
describe("r roles", {
  it("extracts a testthat BDD case", {
    expect_true(TRUE)
  })
})

test_that("extracts a testthat case", {
  expect_equal(1 + 1, 2)
})

test.named.case <- function() {
  TRUE
}

calculate_total <- function() {
  2
}

describe.default("ordinary dotted call", {
  TRUE
})

describe("calculate_total", {
  it("sums", {
    expect_equal(calculate_total(), add_one(1))
  })
})

setup({
  options(demo = TRUE)
})

teardown({
  options(demo = NULL)
})

testthat::test_that("qualified case", {
  expect_true(TRUE)
})

File: fixtures/extraction/lua/test_roles/test_source.lua
describe("lua roles", function()
  before_each(function()
  end)

  it("extracts a busted test case", function()
  end)

  test("aliases a test case", function()
  end)

  pending("waits for a fix")

  strict_teardown(function()
  end)

  insulate("isolated block", function()
    spec("runs in isolation", function()
    end)
  end)
end)

TestCalc = {}

function TestCalc:setUp()
end

function TestCalc:tearDown()
end

function TestCalc:testAdd()
end

function test_named_case()
end

function calculate_total()
  return 2
end

local ordinary = {}
ordinary.it = function(_, callback)
  callback()
end
ordinary.it("ordinary member call", function()
end)
